src/Simplex_tree/include/gudhi/Simplex_tree.h

```diff
@@ -1489,7 +1590,7 @@ class Simplex_tree {
     {
       Node& node_u = static_cast<Node&>(node_as_hook); //corresponding node, has label u
       Simplex_handle sh_u = simplex_handle_from_node(node_u);
-      Siblings * sib_u = self_siblings(sh_u);
+      Siblings* sib_u = const_cast<Siblings*>(self_siblings(sh_u));
```

Comment:
Needing const_cast here looks ugly. Maybe self_siblings should still return a non-const pointer, at least if the Simplex_tree is not const?